tests/tests_fabric/utilities/test_throughput_monitor.py

```diff
@@ -0,0 +1,94 @@
+from unittest import mock
+from unittest.mock import Mock, call
+
+import torch
+from lightning.fabric import Fabric
+from lightning.fabric.utilities.throughput_monitor import ThroughputMonitor, measure_flops
+
+from tests_fabric.helpers.runif import RunIf
+from tests_fabric.test_fabric import BoringModel
+
+
+@RunIf(min_torch="2.1")
+def test_measure_flops():
+    with torch.device("meta"):
+        model = BoringModel()
+        x = torch.randn(2, 32)
+    model_fwd = lambda: model(x)
+    model_loss = lambda y: y.sum()
+
+    training_flops = measure_flops(model, model_fwd, model_loss)
+    assert isinstance(training_flops, int)
+
+    eval_flops = measure_flops(model.eval(), model_fwd, model_loss)
+    assert isinstance(eval_flops, int)
+    assert eval_flops < training_flops
```

Comment:
How is this possible? BoringModel has a linear layer, no dropout or batch norm. model.eval() should not affect any of the forward flops!

Note that we shouldn't confuse `torch.no_grad` with `eval()`, the purpose is very different and so this is unexpected behavior to me. I would agree if the test did

```py
with torch.no_grad():
    eval_flops = measure_flops(model.eval(), model_fwd, model_loss)
assert eval_flops < training_flops
```